core/app/models/spree/promotion_handler/coupon.rb

```diff
@@ -40,13 +43,13 @@ def remove
       end
 
       def set_success_code(status_code)
-        @status_code = status_code
         @success = I18n.t(status_code, scope: 'spree')
+        @success_status_code = status_code
       end
 
       def set_error_code(status_code, options = {})
-        @status_code = status_code
-        @error = options[:error] || I18n.t(status_code, scope: 'spree')
+        @errors.clear
```

Comment:
Can't we use `errors` instead of `@errors` now that we have the `attr_reader`? Also I can't find what `.clear` does in the [`ActiveModel::Errors` documentation](https://www.rubydoc.info/docs/rails/ActiveModel/Errors). Can you help me understand?

Reply:
Does exactly what the name suggests, removes all errors: https://apidock.com/rails/v6.0.0/ActiveModel/Errors/clear

Reply:
Thanks, not sure why it's not present in the link I posted. A couple of things here:

1. I think my first comment is still valid. I think it's better to hide the instance variable here, there's no reason to use it if we have the `attr_reader`.
2. I'm not sure to get why we need to clear all the other errors when setting a new one. I mean, if I need to set multiple errors for a promotions handling and I call `set_error_code` twice, does the second `set_error_code` remove the first error just set? This is the core of this change and specs are successful so I suspect I'm just missing something. 🙂